Automattic/jetpack
# Plans: Add Scan Product to the plans page

#### Changes proposed in this Pull Request:
* Refactors the way that the plans page displays the plans page. 
* Adds a single array that is defined in PHP that then gets used to display the products on the plans page. 
* This hopeful make for a smaller JS bundle size. since the prodct definitions don't live in the JS. 
* It make it much easier to add new products down the line. By just adding to the array. 
* It also make it possible easier for different hosts/ partners to offer different products. The products array can be filterable. 

New Plans Page. 
After the change the pages look like this. 

When the site has no products:
<img width="1107" alt="Screen Shot 2020-04-28 at 1 05 32 PM" src="https://user-images.githubusercontent.com/115071/80481012-f2f94480-8951-11ea-9c75-4acb6034a8a8.png">

When the site has Jetpack Scan product and the Jetpack Premium. 
<img width="1063" alt="Screen Shot 2020-04-28 at 12 55 42 PM" src="https://user-images.githubusercontent.com/115071/80481092-17552100-8952-11ea-80a7-89f3662eacea.png">

The new pop up that was added in https://github.com/Automattic/jetpack/pull/15558 after the rebase.
<img width="766" alt="Screen Shot 2020-04-28 at 12 52 08 PM" src="https://user-images.githubusercontent.com/115071/80481170-3bb0fd80-8952-11ea-8cf4-e22ebc1dfda2.png">
 
#### Testing instructions:
Visit the Plans page. 
- Does it work for you as before. 
- Does it look like before?
There should be no changes in that regard (expect for the new Scan Product)
- Do you see the new Scan Product. 


#### Proposed changelog entry for your changes:
* Adds the new Scan Product to the Plans page.
